wp-parsely.php

```diff
@@ -22,1815 +22,4 @@
-					});
-					return true;
-				}
-			}
-		</script>
-		<script data-cfasync="false" id="parsely-cfg" data-parsely-site="' . esc_attr( $options['apikey'] ) . '" src="https://cdn.parsely.com/keys/' . esc_attr( $options['apikey'] ) . '/p.js"></script>
-		<!-- END Parse.ly Include: Standard -->';
-
-		$registry[ $identifier ] = array(
-			'name'    => $display_name,
-			'payload' => $embed_code,
-		);
-
-		return $embed_code;
-	}
-
-	/**
-	 * Add amp actions.
-	 */
-	public function parsely_add_amp_actions() {
-		if ( ! function_exists( 'is_amp_endpoint' ) || ! is_amp_endpoint() ) {
-			return '';
-		}
-
-		$options = $this->get_options();
-
-		if ( $options['disable_amp'] ) {
-			return '';
-		}
-
-		add_filter( 'amp_post_template_analytics', array( $this, 'parsely_add_amp_analytics' ) );
-		add_filter( 'amp_analytics_entries', array( $this, 'parsely_add_amp_native_analytics' ) );
-	}
-
-	/**
-	 * Add amp analytics.
-	 *
-	 * @param type $analytics The analytics object you want to add.
-	 * @return type
-	 */
-	public function parsely_add_amp_analytics( $analytics ) {
-		$options = $this->get_options();
-
-		if ( empty( $options['apikey'] ) ) {
-			return $analytics;
-		}
-
-		$analytics['parsely'] = array(
-			'type'        => 'parsely',
-			'attributes'  => array(),
-			'config_data' => array(
-				'vars' => array(
-					'apikey' => $options['apikey'],
-				),
-			),
-		);
-
-		return $analytics;
-	}
-
-	/**
-	 * Add amp native analytics.
-	 *
-	 * @param type $analytics The analytics object you want to add.
-	 * @return string|type
-	 */
-	public function parsely_add_amp_native_analytics( $analytics ) {
-		$options = $this->get_options();
-
-		if ( ! empty( $options['disable_amp'] ) && true === $options['disable_amp'] ) {
-			return '';
-		}
-
-		if ( empty( $options['apikey'] ) ) {
-			return $analytics;
-		}
-
-		$analytics['parsely'] = array(
-			'type'       => 'parsely',
-			'attributes' => array(),
-			'config'     => wp_json_encode(
-				array(
-					'vars' => array(
-						'apikey' => $options['apikey'],
-					),
-				)
-			),
-		);
-
-		return $analytics;
-	}
-
-	/**
-	 * Check to see if parsely user is logged in
-	 */
-	public function parsely_is_user_logged_in() {
-		// can't use $blog_id here because it futzes with the global $blog_id.
-		$current_blog_id = get_current_blog_id();
-		$current_user_id = get_current_user_id();
-		return is_user_member_of_blog( $current_user_id, $current_blog_id );
-	}
-
-	/**
-	 * Why is this here?
-	 */
-	public function return_personalized_json() {
-
-	}
-}
-
-
-
-
-if ( class_exists( 'Parsely' ) ) {
-	define( 'PARSELY_VERSION', Parsely::VERSION );
-	$parsely = new Parsely();
-}
-
-require 'class-parsely-recommended-widget.php';
-
-add_action( 'widgets_init', 'parsely_recommended_widget_register' );
-/**
- * Register the Parse.ly Recommended widget.
- */
-function parsely_recommended_widget_register() {
-	register_widget( 'Parsely_Recommended_Widget' );
-}
```

Comment:
This code here is outside of the class, so it should stay in the plugin root file. The new class-parsely.php file should only contain the class, so it can be included without side-effects happening.

Reply:
I'll move it over, but the temptation to refactor at the same time is pretty high :D